src/shared/codelens/localLambdaRunner.ts

```diff
@@ -555,3 +595,43 @@ export async function attachDebugger(params: {
         success: isDebuggerAttached
     }
 }
+
+function recordDebugAttachResult({
+    telemetryService,
+    attachResult,
+    attempts,
+    duration,
+    runtime,
+}: {
+    telemetryService: TelemetryService
+    attachResult: boolean
+    attempts: number
+    duration: number
+    runtime: string
+}): void {
+    const currTime = new Date()
+    const namespace = attachResult ? 'DebugAttachSuccess' : 'DebugAttachFailure'
+
+    const metadata = new Map([
+        ['runtime', runtime],
+    ])
+
+    telemetryService.record({
+        namespace: namespace,
+        createTime: currTime,
+        data: [
```

Comment:
@rli  do we get information like os with the metrics, or do they have to be explicitly set in here?

Reply:
OS is built into the metric

https://github.com/aws/aws-toolkit-vscode/blob/develop/src/shared/telemetry/defaultTelemetryClient.ts#L36